From 5898b844e7c7ed999c0666d3aacb9f0ec3470f9d Mon Sep 17 00:00:00 2001
From: Jakub Adam <jakub.adam@ktknet.cz>
Date: Wed, 17 Feb 2016 14:18:57 +0100
Subject: [PATCH] fix-build-with-glib-2.40

---
 configure.ac     | 4 ++--
 socket/udp-bsd.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6be4010..c7336bd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -93,9 +93,9 @@ AC_CHECK_HEADERS([ifaddrs.h], \
 AC_CHECK_TYPES([size_t, ssize_t])
 
 # Also put matching version in LIBNICE_CFLAGS
-GLIB_REQ=2.44
+GLIB_REQ=2.36
 
-LIBNICE_CFLAGS="-DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_44 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_44"
+LIBNICE_CFLAGS="-DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_36 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_36"
 
 dnl Support different levels of compiler error reporting.
 dnl This configure flag is designed to mimic one from gnome-common,
diff --git a/socket/udp-bsd.c b/socket/udp-bsd.c
index d7346ff..79aa14d 100644
--- a/socket/udp-bsd.c
+++ b/socket/udp-bsd.c
@@ -207,7 +207,7 @@ socket_recv_messages (NiceSocket *sock,
       /* Handle ECONNRESET here as if it were EWOULDBLOCK; see
        * https://phabricator.freedesktop.org/T121 */
       if (g_error_matches (gerr, G_IO_ERROR, G_IO_ERROR_WOULD_BLOCK) ||
-          g_error_matches (gerr, G_IO_ERROR, G_IO_ERROR_CONNECTION_CLOSED))
+          g_error_matches (gerr, G_IO_ERROR, G_IO_ERROR_BROKEN_PIPE))
         recvd = 0;
       else
         error = TRUE;
-- 
2.7.0

